src/components/popover/popover.test.jsx

```diff
@@ -0,0 +1,106 @@
+import React, { useRef, Fragment } from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import Popover from './popover.jsx';
+
+describe('Popover', () => {
+  function PopoverWithToggle(popoverProps) {
+    const popoverRef = useRef();
+
+    return (
+      <Fragment>
+        <button onClick={() => { popoverRef.current.open = true; }}>Open Popover</button>
+        <button onClick={() => { popoverRef.current.open = false; }}>Close Popover</button>
+        <span>Some text</span>
+        <Popover ref={popoverRef} {...popoverProps}>{popoverProps.children}</Popover>
+      </Fragment>
+    );
+  }
+
+  it('has defaults', () => {
+    const popoverRef = React.createRef();
+
+    render(<Popover title="Defaults" />);
+    expect(document.body).toMatchSnapshot();
+
+    render(<Popover title="Defaults" startOpen={true} />);
+    expect(document.body).toMatchSnapshot();
+
+    render(<Popover ref={popoverRef} title="Defaults with ref" />);
+    expect(popoverRef.current).toMatchSnapshot();
+  });
+
+  it('has a title', () => {
+    render(<PopoverWithToggle title="Another title" />);
+
+    userEvent.click(screen.getByText('Open Popover'));
+    expect(screen.getByText('Another title').tagName).toEqual('H1');
+  });
+
+  it('renders children', () => {
+    render(<PopoverWithToggle title="Another title"><span>More text</span></PopoverWithToggle>);
+
+    userEvent.click(screen.getByText('Open Popover'));
+    expect(screen.getByText('More text'));
+  });
+
+  it('can be toggled open/closed', () => {
+    render(<PopoverWithToggle title="Another title" />);
+
+    userEvent.click(screen.getByText('Open Popover'));
+    expect(screen.getByText('Another title')).toBeInTheDocument();
+
+    userEvent.click(screen.getByText('Close Popover'));
+    expect(screen.queryByText('Another title')).not.toBeInTheDocument();
+  });
+
+  it('can be closed by clicking elsewhere in the window', () => {
+    render(<PopoverWithToggle title="Another title" />);
+
+    userEvent.click(screen.getByText('Open Popover'));
+    expect(screen.getByText('Another title')).toBeInTheDocument();
+
+    userEvent.click(screen.getByText('Some text'));
+    expect(screen.queryByText('Another title')).not.toBeInTheDocument();
+  });
+
+  it('can be closed by focusing outside', () => {
+    render(<PopoverWithToggle title="Another title" />);
+
+    userEvent.click(screen.getByText('Open Popover'));
+
+    fireEvent.focusIn(screen.getByText('Close Popover'));
```

Comment:
It's more user-centric to use user actions to perform the focus switching. Something like:

```suggestion
    userEvent.tab();
```

Reply:
Interesting note: it seems that in test land, the whole `window` object can be focused, so we actually have to tab twice to get back to the button. 😅

Reply:
This may also explain the odd behavior with `focus` events occurring with an `event.target` that doesn't inherit from `Node`.